--- dbus-1.9.10/bus/main.c.org	2015-03-01 15:21:04.175051841 -0500
+++ dbus-1.9.10/bus/main.c	2015-03-01 15:28:22.499572030 -0500
@@ -29,6 +29,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <malloc.h>
 #ifdef HAVE_SIGNAL_H
 #include <signal.h>
 #endif
@@ -644,6 +645,13 @@
 #endif /* DBUS_UNIX */
 
   _dbus_verbose ("We are on D-Bus...\n");
+  /*
+   * during init we may have allocated and then freed memory, which
+   * the C library will buffer for us. This is a good place to really
+   * free that memory. Any memory we allocate/free during our normal
+   * operation will still use the C library buffers.
+   */
+  malloc_trim(0);
   _dbus_loop_run (bus_context_get_loop (context));
 
   bus_context_shutdown (context);
