--- dbus-1.9.10/dbus/dbus-mainloop.c~	2015-02-03 15:46:53.000000000 +0000
+++ dbus-1.9.10/dbus/dbus-mainloop.c	2016-05-11 12:51:48.924471251 +0000
@@ -23,6 +23,7 @@
  */
 
 #include <config.h>
+#include <malloc.h>
 #include "dbus-mainloop.h"
 
 #ifndef DOXYGEN_SHOULD_SKIP_THIS
@@ -881,8 +882,10 @@
   _dbus_verbose ("Running main loop, depth %d -> %d\n",
                  loop->depth - 1, loop->depth);
   
-  while (loop->depth != our_exit_depth)
+  while (loop->depth != our_exit_depth) {
+    malloc_trim(0);
     _dbus_loop_iterate (loop, TRUE);
+  }
 
   _dbus_loop_unref (loop);
 }
